src/stores/marker.store.ts

```diff
@@ -1,35 +1,50 @@
 import { Trade } from 'ccxt'
 import { dbDriver, Marker } from '@lib'
+import { StoreOpts } from '@m8bTypes'
+
+const singleton = Symbol()
+const singletonEnforcer = Symbol()
 
 export class MarkerStore {
+  public static get instance(): MarkerStore {
+    if (!this[singleton]) this[singleton] = new MarkerStore(singletonEnforcer)
+    return this[singleton]
+  }
+
   private markers: Map<string, Marker> = new Map()
 
-  public async getNextBackMarker(exchange: string, symbol: string) {
-    const marker = this.getMarker(exchange, symbol)
+  constructor(enforcer: Symbol) {
```

Comment:
Mark this constructor as private, its will also enforce on compiler level, that its only accessible via instance call.